lib/rabbitmq/cli/core/helpers.ex

```diff
@@ -118,23 +119,30 @@ defmodule RabbitMQ.CLI.Core.Helpers do
   defp try_load_rabbit_plugins(opts) do
     with {:ok, plugins_dir} <- plugins_dir(opts)
     do
-      plugins_dir
-      |> File.ls!()
-      |> Enum.filter_map(
-          fn(filename) -> String.ends_with?(filename, [".ez"]) end,
-          fn(archive) ->
-            case Regex.named_captures(~r/(?<name>.+)-(?<version>.+).ez/, archive) do
-              %{"name" => app_name, "version" => _} ->
-                app_dir = Path.basename(archive, ".ez")
-                Path.join([plugins_dir, app_dir, "ebin"]) |> Code.append_path()
-                app_name |> String.to_atom() |> Application.load()
-              _ -> :ok
-            end
-          end)
+      String.split(to_string(plugins_dir), separator())
+      |>
+      Enum.map(&try_load_plugins_from_directory/1)
       :ok
     end
   end
 
+  defp try_load_plugins_from_directory(directory_with_plugins_inside_it) do
+    with {:ok, files} <- File.ls(directory_with_plugins_inside_it)
+    do
+        Enum.filter_map(files,
+            fn(filename) -> String.ends_with?(filename, [".ez"]) end,
+            fn(archive) ->
+              case Regex.named_captures(~r/(?<name>.+)-(?<version>.+).ez/, archive) do
```

Comment:
The code above is used to load plugins code in the command line Elixir app (note it doesn't use the `version` variable). The CLI does a RPC call to `rabbit_plugins:list/1`, which indeed unpacks the EZ file to fetch the version from the `src` file.